internal/handler/handler.go

```diff
@@ -135,8 +137,32 @@ func (h *Handler) Run(ctx context.Context) {
 
 // Handle processes a request.
 func (h *Handler) Handle(ctx context.Context, req *middleware.Request) (*middleware.Response, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/5046
-	panic("not implemented")
+	switch {
+	case req.OpMsg != nil:
+		doc, err := req.OpMsg.Document()
```

Comment:
Ideally decoded `OpMsg.Document()` could be in `*middleware.Request`, but that would require some refactoring so not in this issue.

Reply:
No, `OpMsg.Document()` itself should cache the result. See https://github.com/FerretDB/wire/issues/94

Reply:
Excellent thanks!